File: android/app/src/main/java/com/gooseco/myliftsquad/data/api/OplModels.kt
package com.gooseco.myliftsquad.data.api

data class OplAthlete(
    val name: String,
    val slug: String,
    val country: String?,
    val federation: String?,
    val bestSquat: Double?,
    val bestBench: Double?,
    val bestDeadlift: Double?,
    val bestTotal: Double?,
    val weightClass: String?,
    val equipment: String?,
    val lastCompDate: String?,
    val gender: String?
)

data class CompetitionResult(
    val date: String,
    val meetName: String,
    val federation: String?,
    val equipment: String?,
    val division: String?,
    val weightClassKg: String?,
    val bodyweightKg: Double?,
    val best3SquatKg: Double?,
    val best3BenchKg: Double?,
    val best3DeadliftKg: Double?,
    val totalKg: Double?,
    val place: String?,
    val dots: Double?,
    val meetCountry: String?,
    val meetTown: String?
)

File: ios/MyLiftSquad/Sources/Data/API/OplModels.swift
import Foundation

struct OplAthlete {
    let name: String
    let slug: String
    let country: String
    let federation: String
    let bestSquatKg: Double
    let bestBenchKg: Double
    let bestDeadliftKg: Double
    let bestTotalKg: Double
    let weightClass: String
    let equipment: String
    let lastCompDate: String
    let gender: String
}

struct CompetitionResult: Identifiable {
    let id = UUID()
    let date: String
    let meetName: String
    let federation: String
    let equipment: String
    let division: String
    let weightClassKg: String
    let bodyweightKg: Double
    let best3SquatKg: Double
    let best3BenchKg: Double
    let best3DeadliftKg: Double
    let totalKg: Double
    let place: String
    let dots: Double
    let meetCountry: String
    let meetTown: String
}
